src/librustdoc/html/markdown.rs

```diff
@@ -493,7 +495,15 @@ impl LangString {
                     data.compile_fail = true;
                     seen_rust_tags = true;
                     data.no_run = true;
-                },
+                }
+                x if allow_compile_fail && x.starts_with("E") && x.len() == 5 => {
```

Comment:
I think this should be a separate 'feature gate'. Let's not just tie it to `compile-fail` but add another feature like `check_error_codes`.